test/sql/cast/test_bit_cast.test

```diff
@@ -0,0 +1,216 @@
+# name: test/sql/cast/test_bit_cast.test
+# description: Test casting from and to BitString
+# group: [cast]
+
```

Comment:
Can we add tests using NULL, both to and from BIT?
I believe these should all be possible as every type is nullable

Reply:
To NULL appears to not be possible for types.
Example:
```sql
SELECT 1::INT::NULL;
```
Gives me:
```
Error: Parser Error: syntax error at or near "NULL"
LINE 1: SELECT 1::INT::NULL;
```

Reply:
I think that's expected, NULL isn't a type, it's a constant to represent a null value

Reply:
Added the tests and it behaves as (I would have) expected.